### DISCUSSION TITLE
Ideas and Brainstorming

### DISCUSSION THREAD

**Discussion by Impre-visible** [ChocolateApp/Chocolate#7] [Ideas]
## 👋 Welcome to the Chocolate Discussions!
  We’re using Discussions as a place to:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Brainstorm about features.

  To get started, leave a nice comment or share your idea!

**Comment by albjeremias** [comment]
would be really good to make a script getting info about the series/movie using the opensubtitles-api and then automatically organize in the way you say in the README.md you need the files to be organized! ;) 

- https://www.npmjs.com/package/opensubtitles-api
https://github.com/ChocolateApp/Chocolate#files-organizations

**Reply by Impre-visible** [reply]
Yeah, great idea, it was planned to be done. I'm gonna look at the api, and check if it's possible to have one api key for everyone, not sure it will be possible, but let's give it a shot

**Reply by albjeremias** [reply]
Documentation on open subtitles api:
[Getting started](https://opensubtitles.stoplight.io/docs/opensubtitles-api/e3750fd63a100-getting-started)

[DevReadFirst](https://trac.opensubtitles.org/projects/opensubtitles/wiki/DevReadFirst)

[MovieIdentification](https://trac.opensubtitles.org/projects/opensubtitles/wiki/MovieIdentification)

I guess you need an api key.. but i would like to find out how vlsub handles that.. vlsub is a subtitlte downloader for vlc.
https://github.com/exebetche/vlsub

**Reply by Impre-visible** [reply]
ok, the best think to do is not need an api key, cause peoples don't want to. If it's needed we will do that

**Reply by Impre-visible** [reply]
Hi, It's been a while, but the plugin update will allow people to create their own plugins, and a plugin to import the subtitles from opensubtitles will be possible

**Comment by mauvray** [comment]
Hi,
I saw one of your posts on reddit and checked out the repo, did not try the project yet.
I have two questions :
-  Why flask ?
- Any willing to use flask blueprints instead of one big app.py ? (I can help if needed)

**Reply by Impre-visible** [reply]
Flask was used due to render_template, and the fact that I prefer Flask to Django, and yeah, someday we gonna use flask blueprints instead of one big .py file. If you want to help it's with pleasure, but I think the best is to wait a little week, so I can release the v6.0.0 who refactor a lot of things in the app.py

**Reply by mauvray** [reply]
Thank you for the answer, I'll see you later

**Reply by Impre-visible** [reply]
> Thank you for the answer, I'll see you later

In fack you don't have to wait a week, the release is already available

**Reply by mauvray** [reply]
Thanks that was fast !
Do you have any plan to start decomposing the app.py in blueprints and model files ?

**Reply by Impre-visible** [reply]
I'm gonna look at how blurprints/model work, and I will do that. I have no plan cause I'm still a student, so I have some homeworks to do during my holidays. I will do as fast as I can.

**Reply by mauvray** [reply]
If you want I can draft you a small plan on how to break the app.py

**Reply by Impre-visible** [reply]
yeah, with pleasure
